### PR TITLE
Update dev env setup script

### LINKED ISSUE #1141: Ensure the dev_env_setup.sh and installation guides reflect new module structures

### PR DESCRIPTION
**Pull request by mgasner**
Resolves #1141

### CODE REVIEW

File: dev_env_setup.sh [hunk 1]
```diff
@@ -11,38 +11,32 @@
 pip install --upgrade pip
 pip install -r bin/requirements.txt
 
-pip install -e python_modules/dagster
-pip install -r python_modules/dagster/requirements.txt
+pip install -e python_modules/dagster[aws]
```
Comment (schrockn) [inline dev_env_setup.sh]: ?

dagster_aws

Comment (mgasner) [reply dev_env_setup.sh]: Sadly not. We still have https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/setup.py#L83, without which the S3-backed object store etc. won't work

**Comment by schrockn** [comment]
maybe move S3ObjectStore to dagster_aws?

**Comment by schrockn** [comment]
https://github.com/dagster-io/dagster/issues/1206

**Pull request merged**